lib/container_server.go

```diff
@@ -119,7 +119,7 @@ func New(ctx context.Context, config *Config) (*ContainerServer, error) {
 		return nil, err
 	}
 
-	imageService, err := storage.GetImageService(ctx, store, config.DefaultTransport, config.InsecureRegistries, config.Registries)
+	imageService, err := storage.GetImageService(ctx, nil, store, config.DefaultTransport, config.InsecureRegistries, config.Registries)
```

Comment:
I would provide the system context also as an additional parameter for `New()` to make testing more easy.

Reply:
isn't `nil` here the system context itself? you can mock out system context this way

Reply:
The thing is that I am not able to inject my custom system context because it is always `nil`. But this is just a minor thing, we can also leave it as it is. :+1:

Reply:
oh you meant `server.New` not image service

Reply:
could you follow up on it? :)

Reply:
Yes, sure :+1: